### PR TITLE
chore: switch to python-slim

### PR DESCRIPTION
**Pull request by SdgJlbl**
## Related issue

`#` followed by the number of the issue

## Please check if the PR fulfills these requirements

- [ ] If the feature has an impact on the user experience, the [changelog](https://github.com/substra/substrafl/blob/main/CHANGELOG.md) has been updated
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The commit message follows the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ scores, intermediate_state_train, intermediate_state_agg = simulate_experiment(
 - Abstract base class `Node` is replaced by `Protocols`, defined in `substrafl.nodes.protocol.py` ([#185](https://github.com/Substra/substrafl/pull/185))
 - BREAKING: rename `test_data_sample_keys`, `test_tasks` and `register_test_operations`, `tasks` to `data_sample_keys` and `register_operations` in `TestDataNodes` ([#185](https://github.com/Substra/substrafl/pull/185))
 - BREAKING: `InputIdentifiers` and `OutputIdentifiers` move from `substrafl.nodes.node` to `substrafl.nodes.schemas` ([#185](https://github.com/Substra/substrafl/pull/185))
+- Switch to python-slim as base image, instead of substra-tools ([#197](https://github.com/Substra/substrafl/pull/197))
```
Comment (ThibaultFy) [inline CHANGELOG.md:44]: + drop 3.8 support

**Review by ThibaultFy — APPROVED** [review]
I would add in changelog the drop python 3.8 support too

File: CHANGELOG.md (hunk 1 above)
Comment (SdgJlbl) [reply CHANGELOG.md:44]: Actually, dropping 3.8 entierely is going to require a lot of small adjustments, I'll postpone that to after the release.

File: substrafl/remote/register/register.py [hunk 2]
```diff
@@ -32,11 +32,10 @@
 
 # minimal and maximal values of Python 3 minor versions supported
 # we need to store this as integer, else "3.11" < "3.9" (string comparison)
-MINIMAL_PYTHON_VERSION = 8  # 3.8
+MINIMAL_PYTHON_VERSION = 9  # 3.9
```
Comment (ThibaultFy) [inline substrafl/remote/register/register.py:35]: After thinking, should be done in a separate PR

**Review by ThibaultFy — CHANGES_REQUESTED** [review]

File: substrafl/remote/register/register.py (hunk 2 above)
Comment (SdgJlbl) [reply substrafl/remote/register/register.py]: Agreed 👍

**Review by ThibaultFy — APPROVED** [review]

**Comment by Owlfred** [comment]
```
Usage: /e2e [options] [help]

/e2e may appear anywhere as long as it is on its own line

Options:
  --refs <value>                          Extra refs (branch or tag) with format REPO=GIT_REF,REPO=GIT_REF.
  Supported repositories: orchestrator, substra-backend, substra-frontend, substra-tools, substrafl, substra, substra-tests, substra-documentation, substra-ci.
  Example: /e2e --refs substra-backend=some_branch,orchestrator=some_tag (default: {})
  --tests-to-run, --tests <tests-to-run>  Comma-separated list of tests to run. Valid options: sdk, substrafl, doc, frontend, mnist, camelyon or NONE. Frontend tests will run automatically on substrafl, mnist and camelyon if selected. (default: "sdk")
  -h, --help                              display help for command
```

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: tests/remote/register/test_register.py [hunk 3]
```diff
@@ -44,41 +43,6 @@ def test_check_python_version_valid(version):
     register._check_python_version(version)
 
 
-@pytest.mark.parametrize("use_latest", [True, False])
-def test_latest_substratools_image_selection(use_latest, monkeypatch, default_permissions):
-    monkeypatch.setenv("USE_LATEST_SUBSTRATOOLS", str(use_latest))
```
Comment (ThibaultFy) [inline tests/remote/register/test_register.py:49]: We should at least make a card to remeber to delete this legacy code

File: substrafl/remote/register/register.py [hunk 4]
```diff
@@ -35,8 +29,7 @@
 MINIMAL_PYTHON_VERSION = 8  # 3.8
 MAXIMAL_PYTHON_VERSION = 11  # 3.11
 
-_DEFAULT_SUBSTRATOOLS_IMAGE = "ghcr.io/substra/substra-tools:\
-{substratools_version}-nvidiacuda11.8.0-base-ubuntu22.04-python{python_version}"
+_DEFAULT_SUBSTRATOOLS_IMAGE = "python:{python_version}-slim"
```
Comment (ThibaultFy) [inline substrafl/remote/register/register.py:32]:
```suggestion
_DEFAULT_BASE_DOCKER_IMAGE = "python:{python_version}-slim"
```
Worth renaming it

File: tests/remote/register/test_register.py (hunk 3 above)
Comment (ThibaultFy) [reply tests/remote/register/test_register.py:49]: About `USE_LATEST_SUBSTRATOOLS ` I meant

**Comment by thbcmlowk** [comment]
Should we update the version to release and test the `rc` from this branch before merging?
